Sources/Rendering/WebGPU/RenderWindow/index.js

```diff
@@ -244,19 +244,96 @@ function vtkWebGPURenderWindow(publicAPI, model) {
   }
 
-  publicAPI.captureNextImage = (format = 'image/png') => {
+  publicAPI.captureNextImage = ({ format = 'image/png', size = null } = {}) => {
     if (model.deleted) {
       return null;
     }
     model.imageFormat = format;
     const previous = model.notifyStartCaptureImage;
     model.notifyStartCaptureImage = true;
 
+    model.customCaptureSize = size;
+
     return new Promise((resolve, reject) => {
       const subscription = publicAPI.onImageReady((imageURL) => {
-        model.notifyStartCaptureImage = previous;
-        subscription.unsubscribe();
-        resolve(imageURL);
+        if (model.customCaptureSize === null) {
+          model.notifyStartCaptureImage = previous;
+          subscription.unsubscribe();
+          if (model.resizedScreenshotReplacement) {
+            // resize the main canvas back to its original size and show it
+            model.size[0] = model.originalSize[0];
+            model.size[1] = model.originalSize[1];
+
+            // process the resize
+            publicAPI.modified();
+            // restore the saved camera parameters
+            model.renderable.getRenderers().forEach((ren) => {
+              const camera = ren.getActiveCamera();
+              const params = model.activeCamParamMap.get(ren);
+              camera.setFocalPoint(...params.focalPoint);
+              camera.setPosition(...params.position);
+              camera.setParallelScale(params.pScale);
+            });
+            // re-render
+            publicAPI.traverseAllPasses();
+
+            // Clean up the stand-in
+            model.el.removeChild(model.resizedScreenshotReplacement);
+            model.resizedScreenshotReplacement.remove();
+            model.resizedScreenshotReplacement = null;
+            model.activeCamParamMap = null;
+            model.originalSize = null;
+          }
+          resolve(imageURL);
+        } else {
+          // Create a stand-in image overlay while we resize and render
+          const tmpImg = document.createElement('img');
+          tmpImg.width = model.size[0];
+          tmpImg.height = model.size[1];
+          tmpImg.style = {
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+          };
+          tmpImg.src = imageURL;
+          model.resizedScreenshotReplacement = model.el.appendChild(tmpImg);
+
+          // save the dimensions of the main canvas
+          model.originalSize = [model.size[0], model.size[1]];
+
+          // hide it
+          model.canvas.style.display = 'none';
+
+          // resize it
+          model.size[0] = model.customCaptureSize[0];
+          model.size[1] = model.customCaptureSize[1];
+          model.customCaptureSize = null;
+
+          // process the resize
+          publicAPI.modified();
+          // save camera parameters before resetCamera so we can restore
+          // them later
+          model.activeCamParamMap = new WeakMap();
```

Comment:
I would not use a weak map but just rely on the order of the renderer because it really happen in a non-interactive setup.

You should use a private object on the model. 

```
model._screenshot = {
    originalSize: [],
    imagePaceHolder: ...,
    cameras: [],
};
```